code/modules/atmospherics/gasmixtures/gas_types.dm

```diff
@@ -1,34 +1,65 @@
 GLOBAL_LIST_INIT(hardcoded_gases, list(/datum/gas/oxygen, /datum/gas/nitrogen, /datum/gas/carbon_dioxide, /datum/gas/plasma)) //the main four gases, which were at one time hardcoded
 GLOBAL_LIST_INIT(nonreactive_gases, typecacheof(list(/datum/gas/oxygen, /datum/gas/nitrogen, /datum/gas/carbon_dioxide, /datum/gas/pluoxium, /datum/gas/stimulum, /datum/gas/nitryl))) //unable to react amongst themselves
 
-/proc/meta_gas_list()
+/proc/gas_id2path(id)
+	var/list/meta_gas = GLOB.meta_gas_ids
+	if(id in meta_gas)
+		return id
+	for(var/path in meta_gas)
+		if(meta_gas[path] == id)
+			return path
+	return ""
+
+//Unomos - oh god oh fuck oh shit oh lord have mercy this is messy as fuck oh god
+//my addiction to seeing better performance numbers isn't healthy, kids
+//you see this shit, children?
+//i am not a good idol. don't take after me.
+//this is literally worse than my alcohol addiction
+/proc/meta_gas_heat_list()
 	. = subtypesof(/datum/gas)
 	for(var/gas_path in .)
-		var/list/gas_info = new(7)
 		var/datum/gas/gas = gas_path
+		.[gas_path] = initial(gas.specific_heat)
 
-		gas_info[META_GAS_SPECIFIC_HEAT] = initial(gas.specific_heat)
-		gas_info[META_GAS_NAME] = initial(gas.name)
+/proc/meta_gas_name_list()
+	. = subtypesof(/datum/gas)
+	for(var/gas_path in .)
+		var/datum/gas/gas = gas_path
+		.[gas_path] = initial(gas.name)
+
+/proc/meta_gas_visibility_list()
+	. = subtypesof(/datum/gas)
+	for(var/gas_path in .)
+		var/datum/gas/gas = gas_path
+		.[gas_path] = initial(gas.moles_visible)
 
-		gas_info[META_GAS_MOLES_VISIBLE] = initial(gas.moles_visible)
+/proc/meta_gas_overlay_list()
+	. = subtypesof(/datum/gas)
+	for(var/gas_path in .)
+		var/datum/gas/gas = gas_path
+		.[gas_path] = 0 //gotta make sure if(GLOB.meta_gas_overlays[gaspath]) doesn't break
 		if(initial(gas.moles_visible) != null)
-			gas_info[META_GAS_OVERLAY] = new /list(TOTAL_VISIBLE_STATES)
-			for(var/i in 1 to TOTAL_VISIBLE_STATES)
-				gas_info[META_GAS_OVERLAY][i] = new /obj/effect/overlay/gas(initial(gas.gas_overlay), log(4, (i+0.4*TOTAL_VISIBLE_STATES) / (0.35*TOTAL_VISIBLE_STATES)) * 255)
+			.[gas_path] = new /list(FACTOR_GAS_VISIBLE_MAX)
+			for(var/i in 1 to FACTOR_GAS_VISIBLE_MAX)
+				.[gas_path][i] = new /obj/effect/overlay/gas(initial(gas.gas_overlay), i * 255 / FACTOR_GAS_VISIBLE_MAX)
```

Comment:
Please return this to its prior functionality, it changes how gas overlays smooth together